org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/commands/CheckFilePatternCommand.java

```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
+* Copyright (c) 2021 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.extensions.contentmodel.commands;
+
+import java.nio.file.FileSystems;
+import java.nio.file.PathMatcher;
+import java.nio.file.Paths;
+
+import org.eclipse.lemminx.services.IXMLDocumentProvider;
+import org.eclipse.lemminx.services.extensions.commands.ArgumentsUtils;
+import org.eclipse.lemminx.services.extensions.commands.IXMLCommandService.IDelegateCommandHandler;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lsp4j.ExecuteCommandParams;
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
+
+public class CheckFilePatternCommand implements IDelegateCommandHandler {
+
+	public static final String COMMAND_ID = "xml.check.file.pattern";
+	private IXMLDocumentProvider documentProvider;
+
+	public CheckFilePatternCommand(IXMLDocumentProvider documentProvider) {
+		this.documentProvider = documentProvider;
+	}
+
+	@Override
+	public Object executeCommand(ExecuteCommandParams params, SharedSettings sharedSettings,
+			CancelChecker cancelChecker) throws Exception {
+		String pattern = ArgumentsUtils.getArgAt(params, 0, String.class);
+		PathMatcher matcher = FileSystems.getDefault().getPathMatcher("glob:" + pattern);
+		return documentProvider.getAllDocuments().stream().anyMatch(doc -> matcher.matches(Paths.get(doc.getDocumentURI())));
```

Comment:
I was hoping to get all files within the workspace folder(s), but it seems like this isn't possible without sending their URIs over in the data, which doesn't seem right. Seems like we can only compare against opened files.